pkgs/applications/video/handbrake/default.nix

```diff
@@ -1,72 +1,70 @@
 # Upstream distributes HandBrake with bundle of according versions of libraries and patches to them.
 #
-# Derivation patches HandBrake to use our closure.
+# Derivation patches HandBrake to use Nix closure dependencies.
 #
 
-{ stdenv, lib, fetchurl,
-  python2, pkgconfig, yasm, zlib,
-  autoconf, automake, libtool, m4, jansson,
-  libass, libiconv, libsamplerate, fribidi, libxml2, bzip2,
-  libogg, libopus, libtheora, libvorbis, libdvdcss, a52dec,
-  lame, libdvdread, libdvdnav, libbluray,
-  mp4v2, mpeg2dec, x264, x265, libmkv,
-  fontconfig, freetype, hicolor-icon-theme,
-  glib, gtk3, intltool, libnotify,
-  gst_all_1, dbus-glib, udev, libgudev, libvpx,
-  useGtk ? true, wrapGAppsHook ? null, libappindicator-gtk3 ? null,
-  useFfmpeg ? false, libav_12 ? null, ffmpeg ? null,
+{ stdenv, lib, fetchurl, callPackage,
+  # Main build tools
+  python2, pkgconfig, autoconf, automake, cmake, nasm, libtool, m4,
+  # Processing, video codecs, containers
+  ffmpeg-full, libogg, x264, x265, libvpx, libtheora,
+  # Codecs, audio
+  libopus, lame, libvorbis, a52dec, speex, libsamplerate,
+  # Text processing
+  libiconv, fribidi, fontconfig, freetype, libass, jansson, libxml2,
+  # Optical media
+  libdvdread, libdvdnav, libdvdcss, libbluray,
+  useGtk ? true, wrapGAppsHook ? null,
+                 intltool ? null,
+                 glib ? null,
+                 gtk3 ? null,
+                 libappindicator-gtk3 ? null,
+                 libnotify ? null,
+                 gst_all_1 ? null,
+                 dbus-glib ? null,
+                 udev ? null,
+                 libgudev ? null,
+                 hicolor-icon-theme ? null,
   useFdk ? false, fdk_aac ? null
 }:
 
+let
+
+  nv-codec-headers = callPackage ../../../development/libraries/ffmpeg-full/nv-codec-headers.nix { };
```

Comment:
Why are you using callPackage here when you can simply pass in ```nv-codec-headers``` to the main function?

Reply:
Because there is no such package?
It is declared in `pkgs/development/libraries/ffmpeg-full/nv-codec-headers.nix`
Not declared at top level, that is why I think there is not attribute.

Since it used internally only in FFMpeg-full and required for building HandBrake:
```
../libhb/nvenc_common.c:13:10: fatal error: ffnvcodec/nvEncodeAPI.h: No such file or directory
 #include <ffnvcodec/nvEncodeAPI.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [../libhb/module.rules:12: libhb/nvenc_common.o] Error 1
```

That is why I used `nv-codec-headers` the way FFMpeg-full solely uses it, by directly calling a package.

Reply:
Should I expose `nv-codec-headers` as a package?